### ISSUE TITLE
Title field in document table

### ISSUE THREAD

**Issue by baoea** [ppact/processing#345]
Hi @ayyubibrahimi, I saw a row with a very long `title` in `document` table
`Appeal hearing: Scott Bernard regarding his one day suspension. The hearing was originally scheduled for January 14, 2014. Ms. Angers stated that she received a letter from Michael P. Corry, Attorney for the Police Chief, which stated that he and Ms. Allyson Prejean, attorney for Scott Bernard, requested that the hearing be continued without a date. Mr. Corry stated that he and Ms. Prejean are in the process of working out an agreement without the need for a hearing and would advise the Board if there is a need to have the hearing reset. Ms. Angers informed the Board that she verified that Ms. Prejean was in agreement and she was. on 2014-4-29`

Is it okay because `title` is supposed to be short? Thanks

**Comment by ayyubibrahimi** [comment]
Hi @baoea - The `title` is supposed to be short. The `document` table is produced by Tarak's processing pipeline. Any insight @tarakc02?

**Comment by tarakc02** [comment]
Yes I use some heuristics to determine when the "title" of the hearing ends and when the rest of the content begins. It's supposed to cap the length, I will review why that's not happening and report back.

**Comment by tarakc02** [comment]
In some cases I was pulling out too much text when trying to isolate the name of the accused officer. Since the `title` field combines scraped date and officer name information, that was causing the long titles. In cases where we have matched the officer name to the roster UID (which we do successfully even when there's extra matched text), I've swapped the extracted name with the roster one -- this by itself solved a lot. Additionally, I made a couple of the matching rules for name extraction tighter, so they wouldn't match so much extra text. Finally, as a fallback for if a pattern still matches too much text beyond the name, I included logic to truncate names when the extracted text string was too long to be just a name. This should resolve this issue. The title of the hearing in question is now: `Appeal hearing: Scott Bernard on 2014-4-29`
